cypress/page-objects/app/main-table-page.ts

```diff
@@ -0,0 +1,40 @@
+import { CommonSelectors } from '../../../src/components/data-testid/common-selectors.cy'
+import { MainTableDataCy } from '../../../src/components/data-testid/main-table.cy'
+import { BasePage } from '../lib/base-page'
+import { MainTableWidget } from './widgets/main-table.widget'
+
+class MainTablePage extends BasePage {
+  readonly mainTable: MainTableWidget
```

Comment:
Are you sure you wanna use Widget with all page elements? It looks like unnecessary complexity.
How to fix:
1. Don't use widget at all
2. Split to several widgets